src/client.rs

```diff
@@ -387,11 +390,7 @@ impl Client {
         let storage_key = self._storage_key();
 
         debug!("Retrieve from storage with key={:?}", storage_key);
-        let stored_bytes: Vec<u8> = self
-            .storage
-            .retrieve(&storage_key)
-            .unwrap_or(None)
-            .unwrap_or_else(Vec::new);
+        let stored_bytes: Vec<u8> = self.storage.retrieve(&storage_key).unwrap_or(Vec::new());
```

Comment:
The reason `unwrap_or_else()` exists is that it will only call the closure if necessary, which can be more efficient. The new code will unconditionally create a new `Vec`, regardless of whether it will be needed or not.

In this particular case it doesn't actually matter. The `Vec` implementation is guaranteed not to allocate if the capacity is zero, so creating an empty vector basically is the same as zeroing a few bytes on the stack, and even that can probably be optimised away by the compiler, so you can use whatever you prefer here.

Reply:
I realize that clippy tells me about it :) https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
I decided to go with `.unwrap_or_default()`